### PR TITLE
mgmt core, review update

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Update "get" prefix.

Update `PollResult.Error` constructor to take `HttpHeaders` for consistency.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
Currently I am not able to remove `AzureJacksonAdapter` class.

Its constructor calls a protected `simpleMapper` method from `JacksonAdapter`.
```
    public AzureJacksonAdapter() {
        super();
        serializer().registerModule(ManagementErrorDeserializer.getModule(simpleMapper()));
    }
```

And that `simpleMapper` object uses lots of package private classes like `Base64UrlSerializer` etc.

**Comment by weidongxu-microsoft** [comment]
For whether it is possible to get rid of `PollResult.Error`, and replace it with e.g. `HttpResponseException`, my current investigation find that the instance is constructed from `Error` class in `com.azure.core.management.implementation.polling` https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/Error.java, which need to be serialized, therefore cannot include `HttpResponse` required by `HttpResponseException`.

I can use a `HttpResponseException<String>` without `HttpResponse`, but this seems defeat the purpose of the class.

@anuchandy Would you please advise whether we might have better solution?

**Comment by weidongxu-microsoft** [comment]
"Test Linux From Source - Java 11" CI fails because of the "get" prefix change.

File: sdk/core/azure-core-management/src/main/java/com/azure/core/management/polling/PollerFactory.java [hunk 1]
```diff
@@ -28,6 +28,10 @@
  */
 public final class PollerFactory {
 
+    private PollerFactory() {
+
```
Comment (ChenTanyi) [inline sdk/core/azure-core-management/src/main/java/com/azure/core/management/polling/PollerFactory.java:32]: empty line could be removed

**Review by ChenTanyi — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
@srnagar 

Alternatively, I can make `AzureJacksonAdapter` a private class, and provide a factory method to return a `SerializerAdapter` interface, with `AzureJacksonAdapter` as instance. Is it better?

File: sdk/core/azure-core-management/src/main/java/com/azure/core/management/serializer/SerializerFactory.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+     * Gets the singleton instance of the default management serializer adapter.
+     *
+     * @return the default management serializer adapter.
+     */
+    public static synchronized SerializerAdapter createDefaultManagementSerializerAdapter() {
+        if (serializerAdapter == null) {
+            serializerAdapter = new AzureJacksonAdapter();
+        }
+        return serializerAdapter;
+    }
```
Comment (weidongxu-microsoft) [inline sdk/core/azure-core-management/src/main/java/com/azure/core/management/serializer/SerializerFactory.java:29]: I've hide `AzureJacksonAdapter` to implementation package, then use the factory method to get it.

**Comment by weidongxu-microsoft** [comment]
replaced by https://github.com/Azure/azure-sdk-for-java/pull/15447

**Pull request closed without merging**